test/problems/sampling/protein_folding/bead_contacts/test_contact_map_builder.py

```diff
@@ -0,0 +1,296 @@
+            r_contact,
+        ) = contact_map_builder._create_contact_qubits(peptide)
+
+        self.assertEqual(
+            lower_main_upper_main,
+            {
+                1: {
+                    6: 0.5
+                    * (
+                        I
+                        ^ I
+                        ^ I
+                        ^ I
+                        ^ I
+                        ^ I
+                        ^ I
+                        ^ I
+                        ^ I
+                        ^ I
+                        ^ I
+                        ^ I
+                        ^ I
+                        ^ I
+                        ^ I
+                        ^ I
+                        ^ I
+                        ^ I
+                        ^ I
+                        ^ I
+                    )
+                    - 0.5
+                    * (
+                        I
+                        ^ I
+                        ^ I
+                        ^ I
+                        ^ I
+                        ^ I
+                        ^ I
+                        ^ I
+                        ^ I
+                        ^ I
+                        ^ I
+                        ^ I
+                        ^ I
+                        ^ I
+                        ^ Z
+                        ^ I
+                        ^ I
+                        ^ I
+                        ^ I
+                        ^ Z
+                    )
+                }
+            },
+        )
+        self.assertEqual(
+            lower_side_upper_main,
+            {
+                1: {
+                    5: 0.5
+                    * (
+                        I
+                        ^ I
+                        ^ I
+                        ^ I
+                        ^ I
+                        ^ I
+                        ^ I
+                        ^ I
+                        ^ I
+                        ^ I
+                        ^ I
+                        ^ I
+                        ^ I
+                        ^ I
+                        ^ I
+                        ^ I
+                        ^ I
+                        ^ I
+                        ^ I
+                        ^ I
+                    )
+                    - 0.5
+                    * (
+                        I
+                        ^ I
+                        ^ I
+                        ^ I
+                        ^ I
+                        ^ Z
+                        ^ I
+                        ^ I
+                        ^ I
+                        ^ I
+                        ^ I
+                        ^ I
+                        ^ I
+                        ^ I
+                        ^ I
+                        ^ I
+                        ^ I
+                        ^ I
+                        ^ I
+                        ^ Z
+                    )
+                }
+            },
```

Comment:
I think for some cases like this one we should consider using `# fmt: off` and `# fmt: on` comment blocks to exempt them from black's formatting... @woodsp-ibm @manoelmarques @pbark

Reply:
Yes, that would be useful here, thanks.

Reply:
If this really all that better 

```
            {1: {6: 0.5 * (
                        I ^ I ^ I ^ I ^ I ^ I ^ I ^ I ^ I ^ I ^ I ^ I ^ I ^ I ^ I ^ I ^ I ^ I ^ I
                        ^ I) - 0.5 * (
                                I ^ I ^ I ^ I ^ I ^ I ^ I ^ I ^ I ^ I ^ I ^ I ^ I ^ I ^ Z ^ I ^ I
                                ^ I ^ I ^ Z)}},
```                                

While I agree the spread out formatted version is no good, and so suppressing black format seems fine, can we do better than this which is hard to parse

Reply:
I agree with @woodsp-ibm on this. What about smth similar to the following?
```python
{1:
    {5:
        0.5 * (
            I ^ I ^ I ^ I ^ I ^ I ^ I ^ I ^ I ^ I ^
            I ^ I ^ I ^ I ^ I ^ I ^ I ^ I ^ I ^ I
        ) - 0.5 * (
            I ^ I ^ I ^ I ^ I ^ Z ^ I ^ I ^ I ^ I ^
            I ^ I ^ I ^ I ^ I ^ I ^ I ^ I ^ I ^ Z
        )
    }
},
```
It's not perfect but I think it is closer to something useful/easy to read